mirrord/intproxy/src/proxies/incoming.rs

```diff
@@ -456,10 +520,43 @@ impl BackgroundTask for IncomingProxy {
                     },
 
                     (id, TaskUpdate::Message(msg)) => {
-                        let msg = self.get_subscription(id).and_then(|s| s.wrap_response(msg, id.0));
-                        if let Some(msg) = msg {
-                            message_bus.send(msg).await;
-                        }
+                        let Some(PortSubscription::Steal(_)) = self.get_subscription(id) else {
+                            continue;
+                        };
+                        let msg = match msg {
+                            MessageOut::Raw(bytes) => {
+                                ClientMessage::TcpSteal(LayerTcpSteal::Data(TcpData {
+                                    connection_id: id.0,
+                                    bytes,
+                                }))
+                            },
+                            MessageOut::Http(HttpResponseFallback::Fallback(res)) => {
+                                ClientMessage::TcpSteal(LayerTcpSteal::HttpResponse(res))
+                            },
+                            MessageOut::Http(HttpResponseFallback::Framed(res)) => {
+                                ClientMessage::TcpSteal(LayerTcpSteal::HttpResponseFramed(res))
+                            },
+                            MessageOut::Http(HttpResponseFallback::Streamed(mut res)) => {
+                                let mut body = vec![];
+                                match res.internal_response.body.next_frames(false).await {
+                                        Ok(frames) => {
+                                            let _ = frames.frames.into_iter().map(|frame| body.push(frame.into()));
+                                        },
+                                        _ => continue,
```

Comment:
I think we gotta cover this case as well - send `ChunkedResponse::Error` and do cleanup